sw/device/silicon_creator/rom/rom_state.c

```diff
@@ -0,0 +1,79 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/rom/rom_state.h"
+
+#include "sw/device/silicon_creator/lib/cfi.h"
+#include "sw/device/silicon_creator/lib/shutdown.h"
+
+static OT_WARN_UNUSED_RESULT rom_error_t rom_state_get_state_cb(
+    const rom_state_cb_t state_callbacks[], const size_t state_callbacks_cnt,
+    const rom_state_t state, const rom_state_cb_t **state_cb) {
+  // TODO Start from a random index
+  for (size_t idx = 0; idx < state_callbacks_cnt; idx++) {
```

Comment:
I think you can do this already with: https://github.com/lowRISC/opentitan/blob/7f6d9bda4b2aad7ae5c4e191c2b5cf25eff78f55/sw/device/silicon_creator/lib/drivers/rnd.h#L39

Something like:
```suggestion
  uint32_t start = rnd_uint32();
  for (size_t i = 0; i < state_callbacks_cnt; i++) {
    size_t idx = (start + i) % state_callbacks_cnt;
```

Reply:
Thanks for the suggestion @jadephilipoom. Unfortunately, when applying it I see the ROM execution being stuck an not getting and `RND` data back. This happens in the first ROM state execution, as we're basically calling `rnd_uint32()` before even calling `rom_init`. Even though the EDN is already initialized (in https://github.com/sameo/opentitan/blob/topic/rom-vendor-hooks/sw/device/silicon_creator/rom/rom_start.S#L390-L402), we're not getting data back from it.